docs/en/stack/security/troubleshooting.asciidoc

```diff
@@ -392,6 +392,123 @@ Kerberos/SPNEGO debug logging on JVM, add following JVM system properties:
 
+[[trb-security-saml]]
+=== Common SAML issues
+
+*Symptoms:*
+
+. Authentication in Kibana fails and the following error is printed in the
+  Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Cannot find any matching realm for [SamlPrepareAuthenticationRequest{realmName=null,
+assertionConsumerServiceURL=https://my.kibana.url/api/security/v1/saml}]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Cannot find metadata for entity [your:entity.id] in [metadata.xml]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+unable to authenticate user [<unauthenticated-saml-user>]
+for action [cluster:admin/xpack/security/saml/authenticate]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Authentication to realm my-saml-realm failed -
+Provided SAML response is not valid for realm saml/my-saml-realm
+(Caused by ElasticsearchSecurityException[SAML Response is not a 'success' response:
+Code=urn:oasis:names:tc:SAML:2.0:status:AuthnFailed Message=null Detail=null])
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+The XML Signature of this SAML message cannot be validated. Please verify that the saml
+realm uses the correct SAMLmetadata file/URL for this Identity Provider
+
+-----------------------------------------------
+--
+
+. Users are unable to login with a local username/password in Kibana because SAML is
+enabled.
+
+
+*Resolution:*
+
+. This can be resolved by correcting your `sp.acs` setting in your {es} SAML
+realm configuration or by using the `xpack.security.public.*` settings in `kibana.yml`.
+{kib} either uses the `server.host` and `server.port` or the `xpack.security.public.*`
+settings to construct the SAML ACS URL and this URL needs to match exactly the URL that
+is configured in the SAML realm configuration.
+
+NOTE: The two URLs are compared as case-sensitive strings and not as canonicalized URLs.
+
+. We could not find the metadata for the SAML Entity ID `your:entity.id` in the metadata
+file (`metadata.xml`) that has been configured. Either the value of `idp.entity_id` setting
+in the {es} SAML realm configuration is wrong/misspelled, or the wrong metadata file has been
+configured in `idp.metadata.path`.
```

Comment:
I think this needs more - my hope is that I can just use this doc as an answer to forum posts, but as it stands it doesn't go quite far enough to get users over the hump.

I think we need a few more specific steps like:
- The `metadata.xml` file need to be from your IdP 
- The start of the metadata file should have an element like below.  The value of the `entityID` attribute should be configured as the `idp.entity_id` in `elasticsearch.yml`
    ```
    <EntityDescriptor ID="0597c9aa-e69b-46e7-a1c6-636c7b8a8070" entityID="https://saml.example.com/f174199a-a96e-4201-88f1-0d57a610c522/" ...
    ``` 
- This is a case-sensitive string comparison ...